ui/src/app/index.tsx

```diff
@@ -1,5 +1,7 @@
 import * as React from 'react';
-import * as ReactDOM from 'react-dom';
 import {App} from './app';
+import {createRoot} from 'react-dom/client';
 
-ReactDOM.render(<App />, document.getElementById('app'));
+const container = document.getElementById('app');
+const root = createRoot(container!);
+root.render(<App />);
```

Comment:
Reference: https://react.dev/blog/2022/03/08/react-18-upgrade-guide#updates-to-client-rendering-apis